src/System.Management.Automation/engine/Utils.cs

```diff
@@ -34,85 +34,130 @@ namespace System.Management.Automation
     /// </summary>
     internal static class Utils
     {
-        private readonly struct PrimitiveRange
-        {
-            internal readonly BigInteger MinValue;
-            internal readonly BigInteger MaxValue;
+        /// <summary>
+        /// Converts a given double value to BigInteger via Math.Round().
+        /// </summary>
+        /// <param name="d">The value to convert.</param>
+        /// <returns>Returns a BigInteger value equivalent to the input value rounded to nearest integer.</returns>
+        internal static BigInteger AsBigInt(this double d) => (BigInteger)Math.Round(d);
```

Comment:
What about constructor `bigint new(decimal value)`?

Reply:
Hm! Let me see if it treats rounding the same way.

Reply:
@iSazonov that constructor does not round, it simply truncates, so everything from `3.01` all the way up to `3.999...` still rounds down to `3`, so it is unsuitable here.

Reply:
I suppose I could `new BigInteger(Math.Round(d))` but I don't know if that's really any different to a straight cast.

Reply:
The constructor https://source.dot.net/#System.Runtime.Numerics/System/Numerics/BigInteger.cs,133

Reply:
I'm not familiar with the code itself but I checked both `double` and `decimal` constructors and it appears that both of them are simply discarding the value after the decimal point and not factoring it into the calculations. 

I modified the code as you suggest, and it echoes my initial results -- the constructor ignores the portion following the decimal point completely, instead of rounding as is generally expected (or, at least, is how `L` suffix behaved originally). I will opt for `=> new BigInteger(Math.Round(d));`

Reply:
We used previously Convert class and the class does the round - [Convert]::ToInt32(8.6) return 9. So I am ok with Math.Round().